src/main/java/com/megastore/controller/ProductController.java

```diff
@@ -1,24 +1,31 @@
 package com.megastore.controller;
 
 import com.megastore.model.Product;
+import com.megastore.service.ProductService;
+import lombok.AllArgsConstructor;
+import lombok.RequiredArgsConstructor;
 import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.PathVariable;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RestController;
 
-import java.util.ArrayList;
-import java.util.List;
-import java.util.stream.Collectors;
+import java.util.Optional;
 
 @RestController
 @RequestMapping("/api/v1/products")
+@RequiredArgsConstructor
 public class ProductController {
 
-    @GetMapping("/hot")
-    public List<Product> getHotProducts(List<Product> productList) {
-        return productList.stream()
-                .filter(hotProducts -> hotProducts.getIsHotProduct().equals(true))
-                .limit(9)
-                .collect(Collectors.toList());
+    private final ProductService productService;
+
+    @GetMapping("/hot/{count}")
```

Comment:
The count variable will be sent to you through a query. You try to receive the variable through a path. Please read the difference between these methods and what else exists.